Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,192 @@
+## Status Subresource For Config-Based Resources
+
+* **Owners:**
+  * [yp969803](https://github.com/yp969803)
+* **Status:**
+  * `In-Progress`
+* **Related Tickets:**
+  * [#3385](https://github.com/prometheus-operator/prometheus-operator/issues/3335)
+* **Other docs:**
+  * NA
+
+> This proposal describes how we will extend the Prometheus operator configuration Custom
+> Resource Definitions (CRDs) with a Status subresource field.
+
+## Why
+
+This will enhance observability by allowing users to verify whether their configurations have been successfully applied to Workload.
+
+## Pitfalls of the current solution
+
+Prometheus operator allows users to define their observability workloads through "workload" resources like `Prometheus`, `PrometheusAgent`, `AlertManager`. The configuration of these workloads can be done dynamically by orchestrating "configuration" resources like `ServiceMonitor`, `PodMonitor`, `ScrapeConfig`, etc which in turn are used to generate the configuration for the workloads that the Prometheus-Operator manages.
+
+Currently, the status subresource is only implemented for workload resources. The absence of the status subresource for configuration resources makes it difficult to determine the source of the generated configuration of the workload resources. Additionally, there is no straightforward way to observe the reconciliation status of configuration resources. While Kubernetes events are available when a configuration is rejected by a workload, they are not sufficient for ongoing visibility or troubleshooting.
+
+## Goals
+
+* Define the structure of the status subresource for the custom resource definitions
+  * `ServiceMonitor`
+  * `PodMonitor`
+  * `ScrapeConfig`
+  * `Probes`
+  * `PrometheusRule`
+  * `AlertmanagerConfig`
+* Information about number of Up/Down targets in status subresource for PodMonitor, ServiceMonitor, Probes and ScrapeConfig.
```

Comment:
We'll also to detail what we want to report about AlertmanagerConfig and PrometheusRule. There's also the information about which config is deployed to which workload.
```suggestion
* Provide information about the targets being scraped and their status for scrape resources (`PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`).
```